### PR TITLE
Query block doesn t work when nested underneath [sc-6478]

### PR DESCRIPTION
**Pull request by haitrr**
put the rule before code block rule because that rule looks for 4 space padded blocks which sometime can be the query block inside the bullet list.
https://github.com/markdown-it/markdown-it/blob/master/lib/rules_block/code.js

### CODE REVIEW

File: src/lib/markdown/queryblock.ts [hunk 1]
```diff
@@ -112,7 +113,7 @@ export default function queryblock(md: MarkdownIt): void {
 
     state.line = nextLine + (haveEndMarker ? 1 : 0);
 
-    const token = state.push("container_query_block", "code", 0);
+    const token = state.push("container_query_block", "query", 0);
```
Comment (hoantran-it) [inline src/lib/markdown/queryblock.ts:116]: why do we need to change this?

File: src/lib/markdown/queryblock.ts [hunk 2]
```diff
@@ -39,10 +42,8 @@ export default function queryblock(md: MarkdownIt): void {
     const markup = state.src.slice(mem, pos);
     const params = state.src.slice(pos, max);
 
-    if (marker === 0x3b /* ; */) {
```
Comment (hoantran-it) [inline src/lib/markdown/queryblock.ts:42]: why do we remove the condition here?

**Comment by hoantran-it** [comment]
Hi @haitrr could you help to explain a bit more detail, thank you!

**Comment by haitrr** [comment]
the markdown parser parse the blocks base on a set of rules, the list is here
https://github.com/markdown-it/markdown-it/blob/7edd820b57a7018a9886b6b2efacc9bdae20ca98/lib/parser_block.js#L16

we support parsing query blocks from document by inject a custom rule into the rule set.
the parser will parse the blocks base on the order of the rule, 
currently we are putting the query block before the `fence` rule which is after the `code` rule https://github.com/markdown-it/markdown-it/blob/7edd820b57a7018a9886b6b2efacc9bdae20ca98/lib/rules_block/code.js#L1
 
 thus that `code` rule run before our rule and catched the indented block then the `query block` rule is skipped => query block is rendered as `code` block ( <code>{content}</code>}.

**Review by hoantran-it — APPROVED** [review]

**Pull request merged**